src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement = connection.prepareStatement(query,
+                            Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car " + car, e);
+        }
+        return insertDriversToCar(car);
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT model, car_id, name, country, manufacturers_id "
+                + "FROM cars c "
+                + "INNER JOIN cars_drivers cd "
+                + "ON cd.car_id = c.id "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturers_id = m.id "
+                + "WHERE car_id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                            PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarWithoutListOfDrivers(resultSet);
+                car.setDrivers(getDriverList(car.getId()));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT model, car_id, name, country, manufacturers_id, driver_id "
+                        + "FROM cars c "
+                        + "INNER JOIN cars_drivers cd "
+                        + "ON cd.car_id = c.id "
+                        + "INNER JOIN manufacturers m "
+                        + "ON c.manufacturers_id = m.id "
+                        + "WHERE c.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                         PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarWithoutListOfDrivers(resultSet);
+                car.setDrivers(getDriverList(car.getId()));
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriverList(car.getId()));
```

Comment:
u forgot to delete it higher ?)
it's right place to use this method

Reply:
Yea)